### PR TITLE
polish(pu): polish NGU

### PR DESCRIPTION
**Pull request by puyuan1996**
## Description

1. polish the nstep_return/q_nstep_td_error/EpsGreedySampleWrapper to be compatiable with NGU, and add corresponding unittest.
2. polish the null_padding() if template don't have 'action' attribute.
3. change the name of reward model: rnd to rnd-ngu, to avoid the confusion between the rnd reward model in NGU and the conventional RND paper.
4. yapf format for some config.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/minigrid/config/minigrid_ngu_config.py [hunk 1]
```diff
@@ -100,6 +101,5 @@
 create_config = minigrid_ppo_ngu_create_config
 
 if __name__ == "__main__":
-    # or you can enter `ding -m serial -c minigrid_ngu_config.py -s 0`
```
Comment (PaParaZz1) [inline dizoo/minigrid/config/minigrid_ngu_config.py:103]: add corresponding CLI

File: ding/rl_utils/td.py [hunk 2]
```diff
@@ -69,40 +69,51 @@ def view_similar(x: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
...
-    return_ = torch.cat(return_list, dim=0)
+            reward_factor[i] = gamma * reward_factor[i - 1]
+        reward_factor = view_similar(reward_factor, reward)
+        return_tmp = reward.mul(reward_factor).sum(0)
+        if value_gamma is None:
+            return_ = return_tmp + (gamma ** nstep) * next_value * (1 - done)
+        else:
+            return_ = return_tmp + value_gamma * next_value * (1 - done)
+
+    else:  # if gamma is list. for NGU
```
Comment (PaParaZz1) [inline ding/rl_utils/td.py:88]: don't not use else branch directly, you should do like this:
```python
if isinstance(gamma, float):
  pass
elif isinstance(gamma, list):
  pass
else:
  raise TypeError("xxx")
```

File: ding/worker/collector/base_serial_evaluator_ngu.py [hunk 3]
```diff
@@ -192,18 +192,24 @@ def eval(
...
+        prev_action = {i: torch.tensor(-1) for i in range(self._env_num)}  # TODO: self.action_shape
         prev_reward_e = {i: to_tensor(0, dtype=torch.float32) for i in range(self._env_num)}
         with self._timer:
             while not eval_monitor.is_finished():
                 obs = self._env.ready_obs
                 obs = to_tensor(obs, dtype=torch.float32)
-                # policy_output = self._policy.forward(obs,beta_index)
-                policy_output = self._policy.forward(beta_index, obs, prev_action, prev_reward_e)  # TODO action,r_e,r_i
+                # TODO(pu): r_i, reward embeding
+                policy_output = self._policy.forward(beta_index, obs, prev_action, prev_reward_e)
```
Comment (PaParaZz1) [inline ding/worker/collector/base_serial_evaluator_ngu.py:202]: why not `beta_index` belongs to policy, rather than in evaluator/collector, it is not modified in `evaluator/collector`

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Review by PaParaZz1 — APPROVED** [review]
